lib/moment-strftime.js

```diff
@@ -35,16 +35,38 @@
   };
 
   moment.fn.strftime = function (format) {
-    var momentFormat, value;
-
-    momentFormat = format;
-
-    Object.keys(replacements).forEach(function (key) {
-      value = replacements[key];
-      momentFormat = momentFormat.replace("%" + key, value);
-    });
-
-    return this.format(momentFormat);
+	// Extract strftime tokens
+	var pattern = /\%([a-z|A-Z]+)/g;
+	var formatTokens = format.match(pattern);
+	
+	if(formatTokens != null){
+		var parse = [], momentString = "";
+		
+		// Build a string to format
+		for (var i in formatTokens){
+			parse[formatTokens[i]] = replacements[formatTokens[i].replace('%', '')];
+			momentString += replacements[formatTokens[i].replace('%', '')] + "::";
+		}
+		momentString = momentString.substring(0, momentString.length - 2);
+		
+		// Format string and build array, the indexes of the array will be the same as formatTokens
+		var momentResult = this.format(momentString);
+		var parsed = momentResult.split("::");
+		
+		// Replace tokens with the formatted value
+		var result = format;
+		for (var i in formatTokens){
+			result = result.replace(formatTokens[i], parsed[i]);
+		}
+	}
+	else{
+		// There is no tokens to format
+		result = format;
```

Comment:
The scope of `result` is hard to reason about here...  Is it global or from line 57, in the `if` clause?